user/contactmethod/store.go

```diff
@@ -466,3 +477,39 @@ func (db *DB) FindAll(ctx context.Context, userID string) ([]ContactMethod, erro
 
 	return scanAll(rows)
 }
+
+func (db *DB) FindLastStatus(ctx context.Context, cmID string) (*n.MessageStatus, error) {
+	err := validate.UUID("Contact Method ID", cmID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = permission.LimitCheckAny(ctx, permission.User)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := db.db.BeginTx(ctx, nil)
```

Comment:
Since we're running a single query -- I don't think we need a transaction. We also need to be careful of locks around the `outgoing_messages` table since it gets a lot of important activity.

One thing this API change introduces is the ability to run queries unthrottled directly against it. They are selects & short-lived -- but using a transaction turns it into a minimum 3 round trips instead of one.